src/main/haskell/kore/src/Kore/Step/Function/Registry.hs

```diff
@@ -113,11 +118,43 @@ axiomToIdAxiomPatternPair level (asKoreAxiomSentence -> axiom) =
 -- 'ApplicationFunctionEvaluator's
 axiomPatternsToEvaluators
     :: Map.Map (Id level) [EqualityRule level]
-    -> Map.Map (Id level) [ApplicationFunctionEvaluator level]
+    -> Map.Map (Id level) (FunctionEvaluators level)
 axiomPatternsToEvaluators axiomPatterns =
-    -- remove the key if there are no associated function evaluators
-    Map.filter (not . null)
-        (mapMaybe axiomPatternEvaluator <$> axiomPatterns)
+    Map.map
+        (fromMaybe (error "Unexpected Nothing"))
+        -- remove the key if there are no associated function evaluators
+        (Map.filter isJust
+            (Map.map equalitiesToEvaluators axiomPatterns)
+        )
+  where
+    splitEqualities
```

Comment:
Wouldn't `Data.List.partition` do the job well here?
https://hackage.haskell.org/package/base-4.12.0.0/docs/Data-List.html#v:partition